### DISCUSSION TITLE
Feature Request: Supporting Ternary Aggregate Function

### DISCUSSION THREAD

**Discussion by handstuyennn** [duckdb/duckdb#5834] [General]
Hi DuckDB team,

I would like to implement `TernaryFunction` for `AggregateFunction`.  I also want to add `BinaryWindow` and `TernaryWindow` functions too.
Please give me advise for this.

Thanks

**Comment by Mytherin** [comment]
All the `Binary`, `Ternary` and `Unary` functions are just helper methods to make it easier to write functions with a specific number of arguments - they are not necessary and you can implement any function you want without them. 

That being said, if you want to extend the current set of helpers to include ternary for aggregates and binary/ternary for window functions that's fine by me. I would have a look at the current implementations and extend them with more parameters. Remember that the functions serve primarily as syntactic sugar - i.e. they should be easy/nice to work with.
